src/middlewared/middlewared/plugins/pool_/dataset_attachments.py

```diff
@@ -36,15 +36,16 @@ async def attachments(self, oid):
         return []
 
     @private
-    async def attachments_with_path(self, path):
+    async def attachments_with_path(self, path, check_path_child_of_attachment=False):
         result = []
         if isinstance(path, str) and not path.startswith('/mnt/'):
             self.logger.warning('%s: uexpected path not located within pool mountpoint', path)
 
         if path:
+            options = {'check_path_child_of_resource': check_path_child_of_attachment}
```

Comment:
`check_path_child_of_resource` isn't a good name and doesn't actually explain what's being done. I think we should probably do something less wordy and call it something like `check_parent` or something along those lines. @anodos325 what do you think?